sdk/java/pulumi/src/main/java/com/pulumi/deployment/MockDeployment.java

```diff
@@ -0,0 +1,156 @@
+package com.pulumi.deployment;
+
+import com.pulumi.core.Output;
+import com.pulumi.core.TypeShape;
+import com.pulumi.core.internal.annotations.InternalUse;
+import com.pulumi.deployment.CallOptions;
+import com.pulumi.deployment.Deployment;
+import com.pulumi.deployment.InvokeOptions;
+import com.pulumi.deployment.internal.DeploymentImpl;
+import com.pulumi.deployment.internal.DeploymentInstanceHolder;
+import com.pulumi.deployment.internal.DeploymentInternal;
+import com.pulumi.deployment.internal.Runner;
+import com.pulumi.resources.CallArgs;
+import com.pulumi.resources.InvokeArgs;
+import com.pulumi.resources.Resource;
+import com.pulumi.resources.ResourceArgs;
+import com.pulumi.resources.ResourceOptions;
+import com.pulumi.resources.Stack;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.function.Function;
+
+import static java.util.Objects.requireNonNull;
+
+/**
+ * Simple mock of {@link Deployment} that does mostly nothing.
+ */
+@InternalUse
+public class MockDeployment extends DeploymentInstanceHolder implements Deployment, DeploymentInternal {
```

Comment:
not sure if we'll need this class in the end, but right now it does exactly what a mockito mock would do